File: taxi/dmp/spark/spark-over-yt/yt-wrapper/src/main/scala/ru/yandex/spark/yt/wrapper/table/TableAttachmentByteBufferReader.scala
package ru.yandex.spark.yt.wrapper.table

import java.nio.ByteBuffer
import java.util
import java.util.Collections

import ru.yandex.yt.ytclient.proxy.internal.TableAttachmentRowsetReader

class TableAttachmentByteBufferReader extends TableAttachmentRowsetReader[ByteBuffer] {
  override protected def parseMergedRow(bb: ByteBuffer, size: Int): util.List[ByteBuffer] = {
    val res = bb.duplicate
    res.position(bb.position)
    res.limit(bb.position + size)
    bb.position(bb.position + size)
    Collections.singletonList(res)
  }
}

File: yt/java/yt-client/src/main/java/tech/ytsaurus/client/TableAttachmentByteBufferReader.java
package tech.ytsaurus.client;

import java.nio.ByteBuffer;
import java.util.Collections;
import java.util.List;

class TableAttachmentByteBufferReader extends TableAttachmentRowsetReader<ByteBuffer> {
    @Override
    protected List<ByteBuffer> parseMergedRow(ByteBuffer bb, int size) {
        ByteBuffer res = bb.duplicate();
        res.position(bb.position());
        res.limit(bb.position() + size);
        bb.position(bb.position() + size);
        return Collections.singletonList(res);
    }
}
